File: Hash Table/1. Two Sum/two_sum.cpp
class Solution
{
public:
    vector<int> twoSum(vector<int> &nums, int target)
    {
        unordered_map<int, int> hashmap;
        vector<int> indices;

        hashmap[nums[0]] = 0;

        for (int i = 1; i < nums.size(); i++)
        {
            if (hashmap.count(target - nums[i]) > 0)
            {
                indices.push_back(i);
                indices.push_back(hashmap[target - nums[i]]);
                break;
            }
            hashmap[nums[i]] = i;
        }
        return indices;
    }
};

File: Hash Table/1. Two Sum/two_sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashmap = {nums[0]: 0}
        indices = []

        for i in range(1, len(nums)):
            if target - nums[i] in hashmap:
                indices.append(i)
                indices.append(hashmap[target - nums[i]])
                break
            hashmap[nums[i]] = i
        return indices